quickwit/quickwit-janitor/src/garbage_collection.rs

```diff
@@ -186,31 +190,16 @@ pub async fn delete_splits_with_files(
     let mut deleted_split_ids = Vec::new();
     let mut failed_split_ids_to_error = Vec::new();
 
-    let mut delete_splits_results_stream = tokio_stream::iter(splits.into_iter())
-        .map(|split| {
-            let moved_storage = storage.clone();
-            async move {
-                let file_entry = FileEntry::from(&split);
-                let split_filename = quickwit_common::split_file(split.split_id());
-                let split_path = Path::new(&split_filename);
-                let delete_result = moved_storage.delete(split_path).await;
-                if let Some(ctx) = ctx_opt {
-                    ctx.record_progress();
-                }
-                (split.split_id().to_string(), file_entry, delete_result)
-            }
-        })
-        .buffer_unordered(MAX_CONCURRENT_STORAGE_REQUESTS);
```

Comment:
Unfortunately, I had to replace this with a more basic task + channel approach as I was unable to convince the compiler about higher order lifetimes which was causing a very hard-to-debug error. I'm not sure why it works with the channel approach and not tokio_stream.